### ISSUE TITLE
Scan of mongo-r4.1.6.tar.gz has unexpected problems

### ISSUE THREAD

**Issue by DennisClark** [nexB/scancode-toolkit#1331]
A scan of mongo-r4.1.6.tar.gz
at https://github.com/mongodb/mongo/archive/r4.1.6.tar.gz 
has some unexpected problems: 

```
Some files failed to scan properly:
Path: mongo-r4.1.6.tar.gz-extract/mongo-r4.1.6/src/third_party/IntelRDFPMathLib20U1/TESTS/readtest.in
Path: mongo-r4.1.6.tar.gz-extract/mongo-r4.1.6/src/third_party/IntelRDFPMathLib20U1/TESTS/readtest.in.OLD
Path: mongo-r4.1.6.tar.gz-extract/mongo-r4.1.6/src/third_party/sqlite-amalgamation-3190300/sqlite/sqlite3.c
```
The failures on the *.in and *.in.OLD files make sense, but I was suprised to see the failure on `sqlite3.c`  which is presumably a standard C language file. 

Complete log: 
```
Extracting archives...
Extracting done.
Setup plugins...
Collect file inventory...
Scan files for: info, licenses, copyrights, packages with 3 process(es)...
Scanning done.
Some files failed to scan properly:
Path: mongo-r4.1.6.tar.gz-extract/mongo-r4.1.6/src/third_party/IntelRDFPMathLib20U1/TESTS/readtest.in
Path: mongo-r4.1.6.tar.gz-extract/mongo-r4.1.6/src/third_party/IntelRDFPMathLib20U1/TESTS/readtest.in.OLD
Path: mongo-r4.1.6.tar.gz-extract/mongo-r4.1.6/src/third_party/sqlite-amalgamation-3190300/sqlite/sqlite3.c
Summary:        info, licenses, copyrights, packages with 3 process(es)
Errors count:   3
Scan Speed:     7.99 files/sec. 97.97 KB/sec.
Initial counts: 22718 resource(s): 21147 file(s) and 1571 directorie(s) 
Final counts:   22718 resource(s): 21147 file(s) and 1571 directorie(s) for 253.06 MB
Timings:
  scan_start: 2019-01-14T195657.118162
  scan_end:   2019-01-14T204717.234029
  setup_scan:licenses: 3.06s
  setup: 3.06s
  inventory: 12.15s
  pre-scan:classify: 38.08s
  pre-scan: 38.08s
  scan:packages: 9.17s
  scan: 2645.18s
  post-scan:summary: 181.40s
  post-scan:license-clarity-score: 57.06s
  post-scan:summary-key-files: 11.63s
  post-scan: 250.09s
  output:json: 23.19s
  output: 23.19s
  total: 3055.27s
Removing temporary files...done.
```

**Comment by pombredanne** [comment]
@DennisClark Thank you for the report!

**Comment by pombredanne** [comment]
This is now solved in #1552 with the introduction of a `deadline` in scanners and better handling certain patterns of tokens in text preparation. These three files have very different styles:
- readtest.in and readtest.in.OLD are mostly text data files with hex and digits and are 126, 297 and 126,298 lines big respectively and have no licenses
- the sqlite file contains 141 `public-domain` dedications scattered over 202,920 lines!

Both are now properly scanned without a timeout in #1552

Thanks++
